### DISCUSSION TITLE
The page cannot be displayed because an internal server error has occurred.

### DISCUSSION THREAD

**Discussion by davetapley** [polymind-inc/acmebot#551] [Q&A] [answered]
This has been working great for over a year, but now visiting `/dashboard` I get this (after signing in):

> The page cannot be displayed because an internal server error has occurred.

It happened on v3, and I tried upgrading per [here](https://github.com/shibayan/keyvault-acmebot/discussions/500), but same error. Any ideas? 😬

**Comment by shibayan** [comment, marked as answer]
The error should be logged in Application Insights deployed at the same time. Please check there.

**Reply by davetapley** [reply]
Nothing useful there, zero "Failed requests".

> The page cannot be displayed because an internal server error has occurred.

is on https://func-....azurewebsites.net/.auth/login/aad/callback

So perhaps it's not even getting to the first function? 🤔

**Reply by shibayan** [reply]
Since this is an App Service Authentication error, it appears to be occurring in an area unrelated to Acmebot. It is recommended to reconfigure Authentication.

**Reply by davetapley** [reply]
That was the clue I needed, I got it working again! 🙏🏻 

For others:
1. Under Authentication click [Upgrade from classic Authentication experience](https://user-images.githubusercontent.com/48232/209989858-7a973021-e1ab-4d45-a8b4-b3b87d6582de.png). Wait a minute.
2. Once upgrade is complete click link to **Identity provider**, see:
3. > ❌ A certificate or secret has expired.
4. Click that link to see [expired secret](https://user-images.githubusercontent.com/48232/209989859-d0813be9-fe2c-4cb2-9cf0-54b698e39470.png).
5. Click 'New client secret' and copy **Value**.
6. Go back to Function App, click Configuration.
7. Update `MICROSOFT_PROVIDER_AUTHENTICATION_SECRET` with copied secret value.
8. `/dashboard` now loads ✅

**Reply by DPOGH** [reply]
Thank you @shibayan and @davetapley for the Tips!

took me a while, but finally I managed to renew the Secret.

in the FunctionApp:

1. navigate to _Settings->Authentication_
2. under **Identity Provider** click the (_FUNCTION_NAME_)
3. navigate to _Manage->Certificates & Secrets_
4. create a **New Client Secret**
5. copy the new secret and close this page to go back to the FunctionApp main page
6. Navigate to _Settings->Environment variables_
7. click on `MICROSOFT_PROVIDER_AUTHENTICATION_SECRET`
8. Update `MICROSOFT_PROVIDER_AUTHENTICATION_SECRET` with copied secret value. (do not forget Apply Apply ;-) )
